### PR TITLE
websocket client implementation

### PR DESCRIPTION
**Pull request by fafhrd91**
@asvetlov @kxepal

### CODE REVIEW

**Comment by kxepal** [comment]
Can we have just a `WebSocketResponse.send` and `WebSocketResponse.receive` which hides details about sending data type? Something like:

```
    def send(self, data):
        if self._closing:
            raise RuntimeError('websocket connection is closing')
        if isinstance(data, str):
            binary=False
        elif isinstance(data, (bytes, bytearray, memoryview)):
            binary=True
        else:
            raise TypeError('data argument must be str or byte-ish (%r)' % type(data))
        self._writer.send(data, binary=binary)
```

and `WebSocketResponse.receive_msg` -> `WebSocketResponse.receive` with drop the others? User side could handle both cases in the way it needs while having multiple methods to send and receive different type of data causes hide-and-seek problem when client and server or have to agree on some single method or have to fallback on the other one.

Also `self._closing`, not `self._closed`? Is this flag sign about that connection is closed or some in the middle of closing process?

**Comment by fafhrd91** [comment]
It follows aiohttp.web api, also do you remember explicit is better that implicate.

Regarding closing, websockets close process is slightly more complicated than just drop connection. I am not really happy with exception on close

**Comment by kxepal** [comment]
Ok, I didn't use aiohttp websocket much to judge it API, just this moment caught my eyes. Explicit is better than implicit for sure, but not in way to make simple things complex or we need to talk about `aiohttp.request` (; but that's bikeshedding... Point on current aiohttp.web API is fair one.

LGFM then. Yay websockets! (:

**Comment by fafhrd91** [comment]
`receive_msg()` renamed to `receive()`

**Comment by ludovic-gasc** [comment]
API looks good for me.
I'll try as soon as possible.

File: aiohttp/errors.py [hunk 1]
```diff
@@ -44,6 +46,23 @@ def message(self):
         return self.args[1]
 
 
+class WSServerDisconnectedError(ClientDisconnectedError):
```
Comment (asvetlov) [inline aiohttp/errors.py]: I think parent should be `ServerDisconnectedError`, not `ClientDisconnectedError`.

**Comment by fafhrd91** [comment]
@asvetlov i am merging?

File: aiohttp/websocket_client.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        if not self._closing:
+            self._closing = True
+            self._writer.close(code, message)
+
+    @asyncio.coroutine
+    def wait_closed(self):
+        yield from self._closing_fut
+
+    @asyncio.coroutine
+    def receive(self):
```
Comment (asvetlov) [inline aiohttp/websocket_client.py:120]: I still think the method name should be the same as for server web socket object.
If you prefer `receive` please add it to `WebSocketResponse` with keeping `receive_msg` as alias (think about backward compatibility).

**Comment by asvetlov** [comment]
I have few comments.
Everything else looks perfect!

**Comment by fafhrd91** [comment]
sure, will do. i still want to work with ping/pong and custom close flow.

**Pull request merged**